### PR TITLE
fix: keycloak wait for

### PR DESCRIPTION
**Pull request by Morriz**
multiple issues solved:
- `NODE_EXTRA_CA_CERTS=...` translated to `NODE_TLS_REJECT_UNAUTHORIZED='0'` for keycloak's `openid-client` because it does not seem to honor extra CA
- `404` and `503` now excluded from status codes that stop retrying

### CODE REVIEW

File: src/tasks/keycloak/keycloak.ts [hunk 1]
```diff
@@ -47,17 +47,37 @@ const errors: string[] = []
...
 async function main(): Promise<void> {
-  if (!env.isDev) await waitTillAvailable(env.KEYCLOAK_ADDRESS)
-
+  await waitTillAvailable(env.KEYCLOAK_ADDRESS)
   const keycloakAddress = env.KEYCLOAK_ADDRESS
   const basePath = `${keycloakAddress}/admin/realms`
   let token: TokenSet
+  const isHttps = env.KEYCLOAK_ADDRESS.startsWith('https://')
+  if (isHttps && process.env.NODE_EXTRA_CA_CERTS) {
+    // unfortunately openid client does not seem to honor NODE_EXTRA_CA_CERTS
```
Comment (j-zimnowoda) [inline src/tasks/keycloak/keycloak.ts:56]: I am not sure, that would be also the case for harbor and gitea but we did not observe that, did we?
Would you like to reproduce it together?

File: src/utils.ts [hunk 2]
```diff
@@ -76,8 +78,11 @@ const sleep = (ms) => {
 
 export const waitTillAvailable = async (url: string, opts?: WaitTillAvailableOptions): Promise<void> => {
   const options: WaitTillAvailableOptions = { ...defaultOptions, ...opts }
+  if (env.isDev) options.confirmations = 1
   const isHttps = url.startsWith('https://')
-  const rejectUnauthorized = !(options.skipSsl || !process.env.NODE_TLS_REJECT_UNAUTHORIZED)
+  const globalSkipSsl = !env.NODE_TLS_REJECT_UNAUTHORIZED
```
Comment (j-zimnowoda) [inline src/utils.ts:83]: It is hard for me to find a use case for logic around TLS flags.
Is environment instructs to skip TLS validation (NODE_TLS_REJECT_UNAUTHORIZED) then why a user would like to enforce it. And the other way round. Could you please clarify your understanding of this issue?

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]

**Review by 0-sv — APPROVED** [review]

File: src/tasks/keycloak/keycloak.ts (hunk 1 above)
Comment (Morriz) [reply src/tasks/keycloak/keycloak.ts]: removed as it was unneeded

File: src/utils.ts (hunk 2 above)
Comment (Morriz) [reply src/utils.ts:90]: good point...it was just another version of what marc made but I guess it serves no use...wil remove

**Pull request merged**